velox/functions/lib/window/NthValue.cpp

```diff
@@ -38,14 +37,25 @@ class NthValueFunction : public exec::WindowFunction {
         isConstantOffsetNull_ = true;
         return;
       }
-      constantOffset_ =
-          args[1]
-              .constantValue->template as<ConstantVector<int64_t>>()
-              ->valueAt(0);
+      if (args[1].type->isInteger()) {
+        constantOffset_ =
+            args[1]
+                .constantValue->template as<ConstantVector<int32_t>>()
+                ->valueAt(0);
+      } else {
+        constantOffset_ =
+            args[1]
+                .constantValue->template as<ConstantVector<int64_t>>()
+                ->valueAt(0);
+      }
       VELOX_USER_CHECK_GE(
           constantOffset_.value(), 1, "Offset must be at least 1");
       return;
     }
+
+    VELOX_USER_CHECK(
```

Comment:
Maybe add a variation of this check in the if condition above

if (args[1].type->isInteger()) {
  VELOX_USER_CHECK(args[1].index.has_value(), ""Offset must be literal for spark")
}

so that the logic remains together.

Reply:
Updated, thanks.